bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/read/ResultRowAdapter.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.hbase.adapters.read;
+
+import com.google.api.core.InternalApi;
+import com.google.cloud.bigtable.data.v2.models.RowAdapter;
+import com.google.cloud.bigtable.hbase.util.ByteStringer;
+import com.google.cloud.bigtable.hbase.util.TimestampConverter;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.ByteString;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.client.Result;
+
+/**
+ * Adapter for {@link RowAdapter} that uses {@link Result} to represent logical rows.
+ *
+ * <p>For internal use only - public for technical reasons.
+ */
+@InternalApi("For internal usage only")
+public class ResultRowAdapter implements RowAdapter<Result> {
+
+  @Override
+  public RowBuilder<Result> createRowBuilder() {
+    return new ResultRowBuilder();
+  }
+
+  @Override
+  public boolean isScanMarkerRow(Result result) {
+    return result.isEmpty();
+  }
+
+  @Override
+  public ByteString getKey(Result result) {
+    return ByteStringer.wrap(result.getRow());
```

Comment:
I'm a little concerned about this:
We get a ByteString from a chunk, extract the bytes and then  almost immediately re-wrap it for ReadRowsResumptionStrategy. I need to think about how this can be improved